frontend/public/components/utils/name-value-editor.jsx

```diff
@@ -113,3 +100,153 @@ NameValueEditor.defaultProps = {
+    // but it's good here for the sake of performance
+    // to avoid expensive index searches.
+    monitor.getItem().index = hoverIndex;
+  }
+};
+
+const collectSourcePair = (connect, monitor) => ({
+  connectDragSource: connect.dragSource(),
+  connectDragPreview: connect.dragPreview(),
+  isDragging: monitor.isDragging()
+});
+
+const collectTargetPair = (connect) => ({
+  connectDropTarget: connect.dropTarget(),
+});
+
+const PairElement = DragSource(DRAGGABLE_TYPE.ROW, pairSource, collectSourcePair)(DropTarget(DRAGGABLE_TYPE.ROW, itemTarget, collectTargetPair)(class PairElement extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this._onRemove = this._onRemove.bind(this);
+    this._onChangeName = this._onChangeName.bind(this);
+    this._onChangeValue = this._onChangeValue.bind(this);
+  }
+
+  _onRemove() {
+    const {index, onRemove} = this.props;
+    onRemove(index);
+  }
+
+  _onChangeName(e) {
+    const {index, onChange} = this.props;
+    onChange(e, index, NameValueEditorPair.Name);
+  }
+
+  _onChangeValue(e) {
+    const {index, onChange} = this.props;
+    onChange(e, index, NameValueEditorPair.Value);
+  }
+
+  render() {
+    const {isDragging, connectDragSource, connectDragPreview, connectDropTarget, nameString, valueString, allowSorting, readOnly, pair} = this.props;
+    const deleteButton = <React.Fragment><i className="fa fa-minus-circle pairs-list__btn pairs-list__delete-icon" aria-hidden={true} onClick={this._onRemove}></i><span className="sr-only">Delete</span></React.Fragment>;
+
+    return connectDropTarget(
+      connectDragPreview(
+        <div className={classNames('row', isDragging ? 'pairs-list__row-dragging' : 'pairs-list__row')} ref={node => this.node = node}>
+          <div className="col-xs-5 pairs-list__field">
+            <input type="text" className="form-control" placeholder={nameString.toLowerCase()} value={pair[NameValueEditorPair.Name]} onChange={this._onChangeName} disabled={readOnly}/>
+          </div>
+          <div className="col-xs-6 pairs-list__field">
+            {
+              _.isPlainObject(pair[NameValueEditorPair.Value]) ?
+                <ValueFromPair pair={pair[NameValueEditorPair.Value]}/> :
+                <input type="text" className="form-control" placeholder={valueString.toLowerCase()} value={pair[NameValueEditorPair.Value] || ''} onChange={this._onChangeValue} disabled={readOnly}/>
+            }
+          </div>
+          {
+            readOnly ? null :
+              <div className="col-xs-1">
+                <span className={classNames(allowSorting ? 'pairs-list__span-btns' : null)}>
+                  {
+                    allowSorting ?
+                      <React.Fragment>
+                        {connectDragSource(<i className="fa fa-bars pairs-list__reorder-icon"/>)}
+                        {deleteButton}
+                      </React.Fragment>
+                      :
+                      <React.Fragment>
```

Comment:
I think you can get rid of this <React.Fragment>

Reply:
I can, I just had to get rid of the braces.